From 35e0bf63f7ae8efef9a073adb83915919bfd4cd8 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Thu, 20 Jul 2017 10:58:19 +0800
Subject: [PATCH 2177/5242] MLK-16037: ARM64: dts: switch to use DMA.I2C0
 instead HDMI.I2C0

commit  39051d0c281a325ae13de3d24747b981eb042279 from
https://source.codeaurora.org/external/imx/linux-imx.git

Use the DMA.I2C0 instead HDMI.I2C0, they share same hardware pin
in imx8qm, then the HDMI power domain will not be enabled when
audio codec is working.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../dts/freescale/fsl-imx8qm-lpddr4-arm2-mqs.dts   |    2 +-
 .../dts/freescale/fsl-imx8qm-lpddr4-arm2-spdif.dts |    2 +-
 .../boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts  |   16 ++++++++++++++++
 3 files changed, 18 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-mqs.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-mqs.dts
index fbc4ec3..6a12760 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-mqs.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-mqs.dts
@@ -139,7 +139,7 @@
 	};
 };
 
-&i2c0_hdmi {
+&i2c0 {
 
 	wm8962: wm8962@1a {
 		compatible = "wlf,wm8962";
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-spdif.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-spdif.dts
index 63de79c..85f792e 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-spdif.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-spdif.dts
@@ -142,7 +142,7 @@
 	status = "disabled";
 };
 
-&i2c0_hdmi {
+&i2c0 {
 
 	wm8962: wm8962@1a {
 		compatible = "wlf,wm8962";
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
index 6ce6a33..364e1de 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
@@ -197,6 +197,13 @@
 			>;
 		};
 
+		pinctrl_lpi2c0: lpi2c0grp {
+			fsl,pins = <
+				SC_P_HDMI_TX0_TS_SCL_DMA_I2C0_SCL   0xc600004c
+				SC_P_HDMI_TX0_TS_SDA_DMA_I2C0_SDA   0xc600004c
+			>;
+		};
+
 		pinctrl_lpi2c1: lpi2c1grp {
 			fsl,pins = <
 				SC_P_GPT0_CLK_DMA_I2C1_SCL              0xc600004c
@@ -550,6 +557,15 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_hdmi_lpi2c0>;
 	clock-frequency = <100000>;
+	status = "disabled";
+};
+
+&i2c0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lpi2c0>;
+	clock-frequency = <100000>;
 	status = "okay";
 
 	codec: cs42888@48 {
-- 
1.7.9.5

